src/emulator/pubsubEmulator.ts

```diff
@@ -11,6 +11,14 @@ import { FirebaseError } from "../error";
 import { EmulatorRegistry } from "./registry";
 import { SignatureType } from "./functionsEmulatorShared";
 import { CloudEvent } from "./events/types";
+import { exec } from "child_process";
+
+// Finds processes with "pubsub-emulator" in the description and runs `kill` if any exist
+// Since the pubsub emulator doesn't export any data, force-killing will not affect export-on-exit
+// Note the `[p]` is a workaround to avoid selecting the currently running `ps` process.
+const PUBSUB_KILL_COMMAND =
+  "pubsub_pids=$(ps aux | grep '[p]ubsub-emulator' | awk '{print $2}');" +
+  " if [ ! -z '$pubsub_pids' ]; then kill -9 $pubsub_pids; fi;";
```

Comment:
I'm a little afraid of what this would do on Windows - it's very obviously not going to work (because it's using unix tooling), but will this not cause issues when we try to stop?

Reply:
Good point - I need to get a windows machine to test on. I'll hold off on this righ tnow

Reply:
Added a windows check PTAL